### PR TITLE
Fight page

### PR DESCRIPTION
**Pull request by IlanSirisky**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,9 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.4",
+    "@emotion/styled": "^11.11.5",
+    "@mui/material": "^5.15.18",
```
Comment (oranroz11) [inline package.json]: remove ^

File: src/components/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 import { AutocompleteOptionsType } from "../../types/optionsTypes";
 import AutocompleteListItem from "./AutocompleteListItem";
 
 interface AutocompleteProps {
   options: AutocompleteOptionsType[];
+  inputValue: string;
+  onInputChange: (event: React.SyntheticEvent, value: string) => void;
   placeholder?: string;
+  value?: AutocompleteOptionsType;
   sx?: object;
```
Comment (oranroz11) [inline src/components/Autocomplete/Autocomplete.tsx]: add type

File: src/features/fight/PokemonFightCard/PokemonFightCard.tsx [hunk 3]
```diff
@@ -39,8 +47,25 @@ const PokemonFightCard = ({
...
+        <ThemeProvider theme={healthBarTheme}>
+          {hp && currentHp && (
+            <LinearProgress
+              color={
+                currentHealthPercentage > 50
+                  ? "success"
+                  : currentHealthPercentage > 20
+                  ? "warning"
+                  : "secondary"
+              }
```
Comment (oranroz11) [inline src/features/fight/PokemonFightCard/PokemonFightCard.tsx]: lets create a function - 
`const getColor = (healthPercentage) => {
  if (healthPercentage > 50) {
    return "success";
  } else if (healthPercentage > 20) {
    return "warning";
  } else {
    return "secondary";
  }
};`

File: src/features/fight/PokemonFightCard/styles.ts [hunk 4]
```diff
@@ -1,20 +1,40 @@
...
   gap: ${cssSpacings.s4};
+  height: 40px;
+  width: 70%;
   align-items: center;
-`;
\ No newline at end of file
+`;
+
+export const healthBarStyles = {
+  background: "#c4d6c0",
```
Comment (oranroz11) [inline src/features/fight/PokemonFightCard/styles.ts]: color semantics

File: src/features/fight/PokemonFightCard/styles.ts [hunk 5]
```diff
@@ -1,20 +1,40 @@
...
+export const healthBarTheme = createTheme({
+  palette: {
+    success: {
+      main: "#87c877",
+    },
+    secondary: {
+      main: "#dc004e",
+    },
+    warning: {
+      main: "#ff9800",
```
Comment (oranroz11) [inline src/features/fight/PokemonFightCard/styles.ts]: if that is the palette - so maybe lets make something global for the whole project

File: src/pages/FightPage/FightPage.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        <PokemonFightCard
+          cardTitle={opponentPokemon.name}
+          image={opponentPokemon.imageSrc}
+          hp={opponentPokemon.hp}
+          currentHp={20}
+          playerName="Enemy"
+          subheadText={`${opponentPokemon.id}`}
+          cornerText={`${opponentPokemon.px}px`}
+          topCornerIcon={strengthIcon}
+        />
```
Comment (oranroz11) [inline src/pages/FightPage/FightPage.tsx:125]: to much code in one place - lets separate the buttons types to different components/functions - fight actions and fight button

and try to reduce duplicate code

File: src/pages/FightPage/styles.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+import { COLORS } from "../../styles/colors";
+import { StyledParagraph } from "../../styles/typography";
+import { cssFontWeights, cssSpacings } from "../../styles/stylingValues";
+import yellowBg from "../../assets/yellow-bg.jpeg";
+
+const { Neutrals, Primary } = COLORS;
+
+export const StyledFightPageWrapper = styled(flexColumn)`
+  background-color: ${Neutrals.N100};
+  padding: 40px 40px 80px 40px;
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing attribute

File: src/pages/FightPage/styles.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+export const StyledFightArea = styled(flexRow)`
+  background-image: url(${yellowBg});
+  background-size: cover;
+  background-repeat: no-repeat;
+  width: 100%;
+  height: 100vh;
+  justify-content: space-around;
+  align-items: center;
+  gap: 80px;
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing

File: src/pages/FightPage/styles.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  border-radius: ${cssSpacings.s8};
+`;
+
+export const FightButtonStyle = {
+  fontSize: "48px",
+  lineHeight: "52px",
+  fontWeight: `${cssFontWeights.bold}`,
+  borderRadius: "50%",
+  gap: `${cssSpacings.s10}`,
+  padding: "90px 60px",
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing

File: src/pages/FightPage/styles.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+export const FightActionWrapper = styled(flexColumn)`
+  gap: 32px;
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing

File: src/pages/FightPage/styles.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+`;
+
+export const ActiveButtonStyle = {
+  fontSize: "40px",
+  lineHeight: "52px",
+  fontWeight: `${cssFontWeights.medium}`,
+  padding: "30px 57.5px",
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing

File: src/pages/FightPage/styles.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  borderRadius: "200px",
+  boxShadow: "0px 9px 17.6px 0px #00000040",
```
Comment (oranroz11) [inline src/pages/FightPage/styles.ts]: use spacing and color palette

**Review by oranroz11 — CHANGES_REQUESTED** [review]

File: src/components/Autocomplete/Autocomplete.tsx (hunk 2 above)
Comment (IlanSirisky) [reply src/components/Autocomplete/Autocomplete.tsx]: Didnt understand

**Review by YuvalLeviMoveoHLS — APPROVED** [review]

**Pull request merged**